File: javascript/adicionar-dominio-remetente.js
const https = require('https');

// Definição do endpoint da API
const url = "https://api.mailgrid.net.br/domain/add/";

// Dados de autenticação e domínio a ser adicionado
const data = JSON.stringify({
    "token_auth": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx", // Token de autenticação obtido no painel do cliente
    "dominio": "dominioaseradicionado.net.br" // Domínio a ser adicionado para envios
});

// Configuração da requisição
const options = {
    hostname: 'api.mailgrid.net.br',
    path: '/domain/add/',
    method: 'POST',
    headers: {
        'Content-Type': 'application/json',
        'Content-Length': data.length
    }
};

// Realizando a requisição HTTP
const req = https.request(options, (res) => {
    let responseData = '';

    // Quando os dados forem recebidos
    res.on('data', (chunk) => {
        responseData += chunk;
    });

    // Quando a resposta estiver completa
    res.on('end', () => {
        try {
            // Converte a resposta para JSON e exibe
            const parsedResponse = JSON.parse(responseData);
            console.log("Resposta da API:");
            console.log(JSON.stringify(parsedResponse, null, 2));
        } catch (error) {
            console.log("Erro ao processar a resposta JSON:", error.message);
        }
    });
});

// Em caso de erro na requisição
req.on('error', (error) => {
    console.error("Erro na requisição:", error.message);
});

// Envia os dados no corpo da requisição
req.write(data);

// Finaliza a requisição
req.end();

File: ruby/adicionar-dominio-remetente.rb

require 'net/http'
require 'json'
require 'uri'

# Definição da URL do endpoint da API
url = URI.parse("https://api.mailgrid.net.br/domain/add/")

# Dados de autenticação e domínio a ser adicionado
data = {
	"token_auth" => "SEU_TOKEN_AQUI", # Token de autenticação obtido no painel do cliente
	"dominio" => "dominioaseradicionado.net.br" # Domínio a ser adicionado
}

# Configura a requisição HTTP
http = Net::HTTP.new(url.host, url.port)

# Cria o objeto de requisição POST
request = Net::HTTP::Post.new(url.path, {'Content-Type' => 'application/json'})
request.body = data.to_json # Converte os dados para JSON e coloca no corpo da requisição

# Configura o tempo de timeout
http.read_timeout = 30 # Tempo limite para a requisição em segundos

begin
	# Envia a requisição para o servidor
	response = http.request(request)

	# Verifica o código de status da resposta HTTP
	if response.code.to_i >= 400
	puts "Erro na requisição. Código de status HTTP: #{response.code}"
	else
	# Decodifica a resposta JSON
	parsed_response = JSON.parse(response.body)

	# Verifica se a resposta foi decodificada corretamente
	if parsed_response.is_a?(Hash)
		puts "Resposta da API:"
		puts JSON.pretty_generate(parsed_response) # Exibe a resposta formatada em JSON
	else
		puts "Erro ao processar a resposta JSON."
	end
	end
rescue StandardError => e
	# Captura erros como problemas de conexão ou tempo limite
	puts "Erro na requisição: #{e.message}"
end
